### DISCUSSION TITLE
temporarily switch to user ROOT using xml configuration

### DISCUSSION THREAD

**Discussion by jochenr** [eclipse/jkube#2703] [Q&A] [answered]
Hi,

in a Dockerfile I can do something like this to switch to the root user for building the image and swuícth back at the end

```
FROM registry.access.redhat.com/ubi8/....

USER root

COPY .....

RUN ....

USER 185
```

How can I do the same in the maven plugin xml configuration?
Using the <user> tag I can get the "USER 185" line at the end of the genberated dockerfile

```
<image>
    ....
    <build>
        ...
        <user>185</user>
    </build>
</image>
```

But how can I use ROOT for executing 
```
<runCmds>
    <run>update-ca-trust</run>
</runCmds>
```

?

<assemly> has a <user> subtag, while <runCmds> or <run> do not have a way to specify a user.

I'd apprectiate a hint how to solve this.

thank you & best regards,
Jochen

**Comment by rohanKanojia** [comment]
I'm not sure whether it's possible to generate multiple user fields using plugin configuration. 

JKube also supports building via Dockerfile (see [quickstarts/maven/docker-file-provided](https://github.com/eclipse/jkube/blob/master/quickstarts/maven/docker-file-provided/pom.xml#L116)), Is it possible to switch to using Dockerfile with jkube?

**Reply by manusa** [reply, marked as answer]
This is not possible at the moment:

https://github.com/eclipse/jkube/blob/5da6068d7756784aef9569568ee80f98da97e296/jkube-kit/config/image/src/main/java/org/eclipse/jkube/kit/config/image/build/DockerFileBuilder.java#L111-L138

**Reply by jochenr** [reply]
thank you for clarification👍

does it make sense to you to file a feature request? or is this an advanced use case for that one should stay with Dockerfiles?

**Reply by manusa** [reply]
Yes, if you find this like a useful feature please go ahead.
However this will imply a deep refactor of our current logic, so I'm not sure if we'll have time to implement in the near future.

**Reply by jochenr** [reply]
o.k., I just opened https://github.com/eclipse/jkube/issues/2707

**Comment by jochenr** [comment]
Thank you for the proposal.

Basically it is possible to use a Dockerfile, but actually that's what I was trying to avoid😉.
For me one of the advantages of JKube is that I have all the configuration in one place (the maven pom.xml).

Nevertheless I'll give it a try.
